### DISCUSSION TITLE
How to set random seed.

### DISCUSSION THREAD

**Discussion by JKYUPSYCHE** [lnccbrown/HSSM#384] [Q&A] [answered]
Hi. I want to set random seed for reproducibility of sampling.
I used 'random.seed( )', but I don't think it's working.
How can I set random seed in HSSM?

**Comment by AlexanderFengler** [comment]
Hi @JKYUPSYCHE,

thank you for trying HSSM.

[This](https://discourse.pymc.io/t/how-to-set-a-seed-for-pm-sample/11497) may help. 
Let us know if it doesn't resolve the issue.

Best,
Alex

**Reply by JKYUPSYCHE** [reply]
Thank you Alex, and sorry for the late reply.
I tried, but It's not working yet.
sampling code is here.


rng = np.random.default_rng(42)

model_test = hssm.HSSM(
    model="ddm",
    ...
)

model_test.sample(idata_kwargs=dict(
        log_likelihood=True
    ), random_seed=rng)


This code doesn't cause an error, but the output value is slightly different.

**Comment by digicosmos86** [comment, marked as answer]
Hi @JKYUPSYCHE,

Per [this page](https://bambinos.github.io/bambi/api/Model.html#bambi.Model), and since HSSM is a wrapper around `bambi`, you can use the `random_seed` argument directly when you call model.sample(). This argument is simply an `int` or a list of `int`s, depending on how many cores you are using. In other words, you need to set a random seed for each core that you are using when sampling.

Let us know if this works.

Thanks!
Paul

**Reply by JKYUPSYCHE** [reply]
Hi Paul,
Now It's working! Thank you for you answer.
